pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java

```diff
@@ -0,0 +1,94 @@
+            if (coordinatorId != null) {
+                validateTopicOwnership(TopicName.TRANSACTION_COORDINATOR_ASSIGN.getPartition(coordinatorId),
+                        authoritative);
+                asyncResponse.resume(pulsar().getTransactionMetadataStoreService().getStores()
+                        .get(TransactionCoordinatorID.get(coordinatorId)).getStatus());
+            } else {
+                getPartitionedTopicMetadataAsync(TopicName.TRANSACTION_COORDINATOR_ASSIGN,
+                        false, false).thenAccept(partitionMetadata -> {
+                    if (partitionMetadata.partitions == 0) {
+                        asyncResponse.resume(new RestException(Response.Status.NOT_FOUND,
+                                "Transaction coordinator not found"));
+                        return;
+                    }
+                    List<CompletableFuture<TransactionCoordinatorStatus>> transactionMetadataStoreInfoFutures =
+                            Lists.newArrayList();
+                    for (int i = 0; i < partitionMetadata.partitions; i++) {
+                        try {
+                            transactionMetadataStoreInfoFutures
+                                    .add(pulsar().getAdminClient().transactions().getCoordinatorStatusById(i));
+                        } catch (PulsarServerException e) {
+                            asyncResponse.resume(new RestException(e));
+                            return;
+                        }
+                    }
+                    List<TransactionCoordinatorStatus> metadataStoreInfoList = new ArrayList<>();
+                    FutureUtil.waitForAll(transactionMetadataStoreInfoFutures).whenComplete((result, e) -> {
+                        if (e != null) {
+                            asyncResponse.resume(new RestException(e));
+                            return;
+                        }
+
+                        for (CompletableFuture<TransactionCoordinatorStatus> transactionMetadataStoreInfoFuture
+                                : transactionMetadataStoreInfoFutures) {
+                            try {
+                                metadataStoreInfoList.add(transactionMetadataStoreInfoFuture.get());
+                            } catch (Exception exception) {
+                                asyncResponse.resume(new RestException(exception.getCause()));
+                                return;
+                            }
+                        }
+                        asyncResponse.resume(metadataStoreInfoList);
+                    });
+                }).exceptionally(ex -> {
+                    log.error("[{}] Failed to get transaction coordinator state.", clientAppId(), ex);
+                    resumeAsyncResponseExceptionally(asyncResponse, ex);
+                    return null;
+                });
+            }
+        } else {
+            asyncResponse.resume(new RestException(503, "Broker don't support transaction!"));
```

Comment:
what about "This Broker is not configured with transactionCoordinatorEnabled=true." ?